cloud_formation/lambda/auto-build-test/auto-build-test.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2019 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Lambda to launch ec2-instances """
+import boto3
+
+REGION = 'us-east-1' # region to launch instance.
+AMI = 'ami-456b493a'
+INSTANCE_TYPE = 't2.micro' # instance type to launch.
+
+EC2 = boto3.client('ec2', region_name=REGION)
+
+def lambda_to_ec2(event, context):
+    """ Lambda handler taking [message] and creating a httpd instance with an echo. """
+
+    # bash script to run:
+    #  - boss-manage commands to build stack
+    #  - run tests using the instance's shell
+    #  - send all results to an S3 bucket
+    #  - send out completion SNS
+    #  - set to shutdown the instance automatically in 60 minutes.
+    init_script = """#!/bin/bash
+echo ' '
+echo 'echo "----------------------Init_Script----------------------'
+echo ' '
+
+#Install python3.5 and guide pip to it
+yes | add-apt-repository ppa:fkrull/deadsnakes
+yes | apt-get update
+yes | apt-get install openssh-server
+yes | ufw allow 22
+yes | apt-get install python3.5
+yes | apt-get install zip unzip
+
+yes | apt-get install python3-pip
+
+cd /home/ubuntu/
+
+# Create dummy .aws configs
+mkdir .aws
+cd .aws
+touch credentials
+touch config
+cd ..
+
+#Install git
+yes | apt-get install git
+
+# Compile Packer binary from source
+wget https://releases.hashicorp.com/packer/0.12.0/packer_0.12.0_linux_amd64.zip
+unzip packer_0.12.0_linux_amd64.zip -d packer
+wait
+echo sudo mv packer /usr/bin/
+sudo mv packer /usr/bin/
+echo export PATH="/usr/bin/packer:$PATH"
+export PATH="/usr/bin/packer:$PATH"
```

Comment:
This is definitely needed, I remember spending quite sometime trying to get it to work without adding it to the PATH, and it would not work. I could remove and try again, but I don't think it particularly hurts to leave it there?

Reply:
You're moving the `packer` folder to `/usr/bin`. If you did `sudo mv packer/packer /usr/bin`, you wouldn't need to change the path.

Reply:
Got it!